Neos.Fusion.Form/Resources/Private/Fusion/Root.fusion

```diff
@@ -0,0 +1,82 @@
+    renderer = ${props.content}
+}
+
+prototype(Neos.Fusion.Form:Form) < prototype(Neos.Fusion.Form:FormComponent) {
+  attributes = Neos.Fusion:DataStructure
+  action = Neos.Fusion:UriBuilder
+  method = 'post'
+  enctype = null
+
+  renderer = afx`
+    <form
+      {...props.attributes}
+      action={props.action}
+      method={props.method}
+      enctype={props.enctype}
+    >
+        <div style="display: none">
+
+            <Neos.Fusion:Fragment @if.isNotMainRequest={!request.mainRequest} >
+                <input type="hidden" name={request.argumentNamespace + "[__referrer][@package]"} value={request.parentRequest.controllerPackageKey} />
+                <input type="hidden" name={request.argumentNamespace + "[__referrer][@subpackage]"} value={request.parentRequest.controllerSubpackageKey} />
+                <input type="hidden" name={request.argumentNamespace + "[__referrer][@controller]"} value={request.parentRequest.controllerName} />
+                <input type="hidden" name={request.argumentNamespace + "[__referrer][@action]"} value={request.parentRequest.controllerActionName} />
+                <input type="hidden" name={request.argumentNamespace + "[__referrer][arguments]"} value={Form.argumentsWithHmac(request.parentRequest.arguments)} />
+            </Neos.Fusion:Fragment>
+
+            <Neos.Fusion:Fragment>
+                <input type="hidden" name="__referrer[@package]" value={request.controllerPackageKey} />
+                <input type="hidden" name="__referrer[@subpackage]" value={request.controllerSubpackageKey} />
+                <input type="hidden" name="__referrer[@controller]" value={request.controllerName} />
+                <input type="hidden" name="__referrer[@action]" value={request.controllerActionName} />
+                <input type="hidden" name="__referrer[arguments]" value={Form.argumentsWithHmac(request.arguments, request.argumentNamespace)} />
+            </Neos.Fusion:Fragment>
+
+            <input
+                type="hidden"
+                name={Form.prefixFieldName('__trustedProperties', form.fieldNamePrefix)}
+                value={Form.trustedPropertiesToken(props.content, form.fieldNamePrefix)}
+                />
+
+            <input type="hidden" name="__csrfToken" value={Form.csrfToken()} />
+        </div>
+        {props.content}
+    </form>
+  `
+}
+
+prototype(Neos.Fusion.Form:Textfield)  < prototype(Neos.Fusion.Form:FieldComponent) {
+  attributes = Neos.Fusion:DataStructure
+
+  renderer = afx`
+      <input
+          {...props.attributes}
+          type="text"
+          name={field.name}
+          value={field.value}
+          />
+    `
+}
+
+prototype(Neos.Fusion.Form:Submit) < prototype(Neos.Fusion.Form:FieldComponent) {
+  attributes = Neos.Fusion:DataStructure
+
+  renderer = afx`
+        <input
+          {...props.attributes}
+          type="submit"
+          name={field.name}
+          value={field.value}
```

Comment:
couldn't we just use props.field.value here....

Reply:
not really as we are in some cases use objectAccess to get the value from the form object or fallback to the submitted value if validation errors occured